blocks/utils.py

```diff
@@ -429,3 +429,47 @@ def ipdb_breakpoint(x):
     """
     import ipdb
     ipdb.set_trace()
+
+
+class LambdaIterator(six.Iterator):
```

Comment:
Can we call it `FunctionIterator`, or does the argument need to be a lambda function per se?